Sources/HAP/Server/Configuration.swift

```diff
@@ -7,25 +7,48 @@ fileprivate let logger = getLogger("hap.device")
 func generateIdentifier() -> String {
     do {
         return Data(bytes: try Random.generate(byteCount: 6))
-            .map { String($0, radix: 16, uppercase: false) }
+            .map { String($0, radix: 16, uppercase: true) }
             .joined(separator: ":")
     } catch {
         fatalError("Could not generate identifier: \(error)")
     }
 }
 
+// Generate a valid random setup code, used to pair with the device
+func generateSetupCode() -> String {
+    let n1 = arc4random_uniform(1000)
+    let n2 = arc4random_uniform(100)
+    let n3 = arc4random_uniform(1000)
+    let setupCode = String(format: "%03ld-%02ld-%03ld", n1, n2, n3)
+    
+    if !Device.isValid(setupCode: setupCode) {
+        return generateSetupCode()
+    }
+    return setupCode
+}
+
+// Generate a random four character setup key, used in setupURI and setupHash
+func generateSetupKey() -> String {
```

Comment:
Maybe we should change all these functions into static functions on the Device?

The first version of Swift used top-level functions quite a lot. Later versions reversed that decision. I still mix&match both styles, especially when those functions are somewhat specific to that file, but don't feel quite right being a class member.

Reply:
I agree, I am not a fan of putting anything in global scope, and had originally put these as static in Device, but for this PR I followed your lead with the generateIdentifier() function. I suggest putting this whole class as an extension of Device. The class will never be used outside of Device.